frontend/admin/src/js/components/poolCandidate/PoolCandidatesTable.tsx

```diff
@@ -123,6 +124,22 @@ const viewAccessor = (
   adminRoutes: AdminRoutes,
   intl: IntlShape,
 ) => {
+  if (!getFeatureFlags().applicantSearch) {
+    return (
+      <span data-h2-font-weight="base(700)">
+        {tableViewItemButtonAccessor(
+          adminRoutes.poolCandidateUpdate(candidate.pool.id, candidate.id),
+          intl.formatMessage({
+            defaultMessage: "Edit",
+            id: "aGEisH",
+            description:
+              "Title displayed for the Pool Candidates table Edit column.",
+          }),
+        )}
+      </span>
+    );
```

Comment:
No block, but was curious where getFeatureFlags was needed and so came to be looking at this and wondering why we're not just treating viewAccessor like a component? That way, we avoid creating the sorts of functions we then need to accommodate...? e.g.,

```ts
interface ViewAccessorProps {
  candidate: PoolCandidate;
}
const ViewAccessor = ({ candidate }: ViewAccessorProps) => {
  const { applicantSearch }  = useFeatureFlags();
  const intl = useIntl();
  const adminRoutes = useAdminRoutes();

  if (!applicantSearch) {
    // ...
```

My  rough understanding is that hooks exist to not need to pass props, so I'm unclear why we'd instantiate values from hooks and pass them in.

I'm finding myself wondering if the reason for this architectural decision (accessor function instead of component) was to "allow" the original author to just keep it in the same file without violating our "one file per component" convention. Otherwise, seems it can easily be treated as a functional component? :)

Happy to see this deferred in the interest of a fast merge, but thought I'd ask in-flow of PR

Reply:
Ah. I see what we were working around with these not-component functions:
Refactoring to:
```
accessor: ({ user }) => (
          <PreferredLanguageAccessor language={user?.preferredLang} />
        ),
```
results in a linting error:
>Do not define components during render. React will see a new component type on every render and destroy the entire subtree’s DOM nodes and state (https://reactjs.org/docs/reconciliation.html#elements-of-different-types). Instead, move this component definition out of the parent component “PoolCandidatesTable” and pass data as props. If you want to allow component creation in props, set allowAsProps option to true.eslint[react/no-unstable-nested-components](https://github.com/jsx-eslint/eslint-plugin-react/tree/master/docs/rules/no-unstable-nested-components.md)

Reply:
So, it can be a pure function w/o the need for a hook. The important part was returning an object with keys so you don't need to know all of the feature flags.

## Before

I need to know all of the feature flags and there is no auto suggest or validating of the feature flag so a typo causes it to always be false.

```tsx
const applicantSearch = checkFeatureFlag("FEATURE_APPLICANTSEARCH"); // True
const applicantSearch = checkFeatureFlag("APPLICANTSEARCH"); // False + no hinting as to why
```

## After

With the hook, we get some auto suggest for what feature flags we have available to check and it will error if you use one that does not exist.

```tsx
const flags = useFeatureFlags();
const { applicantSearch } = flags; // Okay 
const { applicantSearch } = flags; // Error 
```

Reply:
Ah, if it's complex to resolve as the warning recommends, I'll retract :)